src/tracks/playabletrack/wavetrack/ui/WaveClipAdjustBorderHandle.cpp

```diff
@@ -135,29 +122,30 @@ class AdjustClipBorder final : public WaveClipAdjustBorderHandle::AdjustPolicy
 
 private:
    std::shared_ptr<WaveTrack> mTrack;
-   std::vector<std::shared_ptr<WaveClip>> mClips;
+   std::shared_ptr<WaveTrack::Interval> mInterval;
    int mDragStartX{ };
    const bool mAdjustingLeftBorder;
    const double mInitialBorderPosition;
+   double mBorderPosition;
    const std::pair<double, double> mRange;
    AdjustHandler mAdjustHandler;
 
    std::unique_ptr<SnapManager> mSnapManager;
    SnapResults mSnap;
 
-   void TrimTo(double t) const
+   void TrimTo(double t)
    {
-      t = std::clamp(t, mRange.first, mRange.second);
-      for(auto& clip : mClips)
-         mAdjustHandler(*clip, t);
+      mBorderPosition = std::clamp(t, mRange.first, mRange.second);
+      for(unsigned channel = 0; channel < mInterval->NChannels(); ++channel)
+         mAdjustHandler(*mInterval->GetClip(channel), mBorderPosition);
```

Comment:
Can you rewrite this as an interval Manor, channel minor iteration, adding member functions to WaveChannelnterval, and eliminating the use of GetClip ?

Reply:
Not sure that I understand that correctly as there are no iterations over intervals in this function, and there is no need to iterate over intervals because we already have a pointer to the interval.
Having an interface that allows trimming of intervals in individual channels would be wrong thing to do, `WaveTrack::Interval` is more appropriate place for this, right?

Reply:
Right, there is only one level of looping.

What I do mean is, hide the use of `GetClip` and making the loop internal to a new member function of `Interval` would be good.